File: fc-chat/demo/part3/ch3/04-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/domain/ChannelId.java
package net.prostars.messagesystem.dto.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public record ChannelId(@JsonValue Long id) {

  @JsonCreator
  public ChannelId {
    if (id == null || id <= 0) {
      throw new IllegalArgumentException("Invalid ChannelId");
    }
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/domain/ChannelId.kt
package com.yscorp.withpush.messagesystem.dto.domain

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

@JvmRecord
data class ChannelId @JsonCreator constructor(@field:JsonValue val id: Long) {
    init {
        require(!(id == null || id <= 0)) { "Invalid ChannelId" }
    }
}
